sdk/go/common/workspace/project.go

```diff
@@ -102,6 +106,8 @@ type Project struct {
 
 	// Options is an optional set of project options
 	Options *ProjectOptions `json:"options,omitempty" yaml:"options,omitempty"`
+
+	Providers map[string]*ProviderOptions `json:"providers,omitempty" yaml:"providers,omitempty"`
```

Comment:
Again based on how we handle version, this might end up being a list rather than a map. 
```
providers:
- name: aws
  version: 4.0
  path: ../../awsv4/bin
- name: aws
  version: 5.0
  path: ../../awsv5/bin
```
Again @danielrbradley to comment